pkg/cmd/attestation/download/download_test.go

```diff
@@ -0,0 +1,186 @@
+
+		actualLineCount, err := countLines(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+
+		expectedLineCount := 2
+		assert.Equal(t, expectedLineCount, actualLineCount)
+
+		err = os.Remove(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+	})
+
+	t.Run("download OCI image attestations successfully", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "oci://ghcr.io/github/test"
+
+		err = RunDownload(&opts)
+		assert.NoError(t, err)
+
+		artifact, err := artifact.NewDigestedArtifact(opts.OCIClient, opts.ArtifactPath, opts.DigestAlgorithm)
+		require.NoError(t, err)
+
+		assert.FileExists(t, fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+
+		actualLineCount, err := countLines(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+
+		expectedLineCount := 2
+		assert.Equal(t, expectedLineCount, actualLineCount)
+
+		err = os.Remove(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+	})
+
+	t.Run("cannot find artifact", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "../test/data/not-real.zip"
+
+		err := RunDownload(&opts)
+		assert.Error(t, err)
+	})
+
+	t.Run("no attestations found", func(t *testing.T) {
+		opts := baseOpts
+		opts.APIClient = api.MockClient{
+			OnGetByOwnerAndDigest: func(repo, digest string, limit int) ([]*api.Attestation, error) {
+				return nil, nil
+			},
+		}
+
+		err := RunDownload(&opts)
+		assert.NoError(t, err)
+
+		artifact, err := artifact.NewDigestedArtifact(opts.OCIClient, opts.ArtifactPath, opts.DigestAlgorithm)
+		require.NoError(t, err)
+		assert.NoFileExists(t, artifact.DigestWithAlg())
+	})
+
+	t.Run("cannot download OCI artifact", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "oci://ghcr.io/github/test"
+		opts.OCIClient = oci.NewReferenceFailClient()
+
+		err := RunDownload(&opts)
+		assert.Error(t, err)
+		assert.ErrorContains(t, err, "failed to digest artifact")
+	})
+
+	t.Run("with missing OCI client", func(t *testing.T) {
+		customOpts := baseOpts
+		customOpts.ArtifactPath = "oci://ghcr.io/github/test"
+		customOpts.OCIClient = nil
+		assert.Error(t, RunDownload(&customOpts))
+	})
+
+	t.Run("with missing API client", func(t *testing.T) {
+		customOpts := baseOpts
+		customOpts.APIClient = nil
+		assert.Error(t, RunDownload(&customOpts))
+	})
+}
+
+func TestCreateJSONLinesFilePath(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "gh-attestation-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	t.Run("with output path", func(t *testing.T) {
+		artifact, err := artifact.NewDigestedArtifact(oci.NewMockClient(), "../test/data/sigstore-js-2.1.0.tgz", "sha512")
+		require.NoError(t, err)
+		path := createJSONLinesFilePath(artifact.DigestWithAlg(), tempDir)
+
+		expectedPath := fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg())
```

Comment:
You might consider using [`path.Join`](https://pkg.go.dev/path#Join) for formulating file paths for its handling of various edge cases but also using operating-specific rules

> _"Join joins any number of path elements into a single path, separating them with slashes. Empty elements are ignored. The result is Cleaned. However, if the argument list is empty or all its elements are empty, Join returns an empty string."_

Reply:
Good catch, these tests use `path.Join` now.